buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy

```diff
@@ -138,9 +138,10 @@ class NodeInfo {
         String esJavaOpts = config.jvmArgs.isEmpty() ? collectedSystemProperties : collectedSystemProperties + " " + config.jvmArgs
         env.put('ES_JAVA_OPTS', esJavaOpts)
         for (Map.Entry<String, String> property : System.properties.entrySet()) {
-            if (property.getKey().startsWith('es.')) {
+            assert property.key.startsWith('es.') == false : "BUILD MISCONFIGURED: Use -Dtests.es.* to pass system properties to ES. -Des.* is not supported explicitly"
+            if (property.key.startsWith('tests.es.')) {
                 args.add("-E")
-                args.add("${property.getKey()}=${property.getValue()}")
+                args.add("${property.key.substring('tests.es.'.size())}=${property.getValue()}")
```

Comment:
Since you switched the syntax for the key from the method syntax `getKey()` to the attribute syntax `key`, can you do the same for the value too?